lib/autoupdate/agent/config.go

```diff
@@ -203,3 +289,36 @@ func (i InstallFlags) String() string {
 	}
 	return "Unknown"
 }
+
+// DirFlag returns the directory path representation of a single InstallFlag flag, or "unknown".
+func (i InstallFlags) DirFlag() string {
+	switch i {
+	case 0:
+		return ""
+	case FlagEnterprise:
+		return "ent"
+	case FlagFIPS:
+		return "fips"
+	}
+	return "unknown"
+}
+
+func (i InstallFlags) MarshalYAML() (any, error) {
+	return i.Strings(), nil
+}
+
+func (i InstallFlags) MarshalJSON() ([]byte, error) {
+	return json.Marshal(i.Strings())
+}
+
+func (i *InstallFlags) UnmarshalYAML(n *yaml.Node) error {
+	var s []string
+	if err := n.Decode(&s); err != nil {
+		return trace.Wrap(err)
+	}
+	if i == nil {
+		return trace.Errorf("nil install flags while parsing YAML")
```

Comment:
```suggestion
		return trace.BadParameter("nil install flags while parsing YAML")
```

Reply:
What's the motivation for using a typed error in these contexts? The YAML parser (or calling code) doesn't look for BadParameter.

Reply:
Convention? We have an error type that is used heavily throughout the code base whose semantics match the use case here. 

I'd be asking the inverse - why wouldn't we want to use it?